### ISSUE TITLE
Test suite execution and reporting order

### ISSUE THREAD

**Issue by stevej2608** [jasmine/jasmine#1643]
Hi, I have two `jasmine` test suites in separate files that I'm running on node in conjunction with `karma` and reporting using `karma-mocha-reporter`. 

I have ` random: false`. Each suite (A & B) runs five async tests in sequence and then completes.

When I run the suites separately the tests run sequentially and complete as expected. With both suites enabled I get one of two outcomes on each test run:

**Outcome #1 (good)**
START:
  IDBCache Timeout Tests
    √ A1. Create entries
    √ A2. Read data back
    √ A3. Sleep...
    √ A4. Confirm entries have timed out
    √ A5. Flush table
  IDBCache Tests
    √ B1. Create cache instance
    √ B2. Read data back
    √ B3. Count Keys
    √ B4. Delete cache entry
    √ B5. Clear cache

** Outcome #2 (puzzling)**
START:
  IDBCache Timeout Tests
    √ A3. Sleep...
    √ A4. Confirm entries have timed out
    √ A5. Flush table
  IDBCache Tests
    √ B1. Create cache instance
    √ B2. Read data back
    √ B3. Count Keys
    √ B4. Delete cache entry
    √ B5. Clear cache
    √ A1. Create entries
    √ A2. Read data back

In outcome #2 the reports for tests A1 & A2 are appended to the end of the report for suite B.

The A3 test is a simple delay between tests A2 and A4:

```
  it("A3. Sleep...", async (done) => {
    await sleep(2100);
    done();
  }, 3000);
```
I'm no sure how to investigate/resolve this.  Is it a problem with karma, jasmine or karma-mocha-reporter? Any suggestions would be very welcome.

Cheers.

**Comment by slackersoft** [comment]
This sounds like it's probably an issue with how Karma is setting and running up your suites with Jasmine. My guess is that it's trying to speed things up by making two Jasmine's (one for the file with A, and another for file B), and running those more or less in parallel. If they are all loaded into the same Jasmine instance, I would expect them to work exactly as you have in your first example output.

Do you have an entire example suite that reproduces the issue to help debug?

Thanks for using Jasmine!

**Comment by sgravrock** [comment]
I'm going to go ahead and close this since we haven't heard anything more in quite some time. If you still need help or think there's something wrong with Jasmine here, feel free to open a new issue with an example suite that reproduces the problem.
